pkg/etcd/tls.go

```diff
@@ -36,20 +38,23 @@ type TLSGetter interface {
 	Config(path string, sc string) (*transport.TLSInfo, error)
 }
 
-type TLSSecretCacher struct {
-	kubeCli kubernetes.Interface
-	tlsMap  map[string]*transport.TLSInfo
-	mutex   sync.Mutex
+type TLSSecret struct {
+	kubeCli      kubernetes.Interface
+	secretLister listerscorev1.SecretLister
 }
 
-func NewTLSSecretGetter(clientbuilder util.ClientBuilder) TLSGetter {
-	var tlsSecretCacher TLSSecretCacher
-	tlsSecretCacher.kubeCli = clientbuilder.ClientOrDie()
-	tlsSecretCacher.tlsMap = make(map[string]*transport.TLSInfo)
-	return &tlsSecretCacher
+func NewTLSSecretGetter(secretInformer informerscorev1.SecretInformer, clientBuilder util.ClientBuilder) *TLSSecret {
```

Comment:
we have defined an abstract interface to get etcd tls config. It is recommended that we implement two secret getters, which are more elegant, such as
TLSSecretGetter
TLSSecretCacheGetter